specification/resources/resource-manager/Microsoft.Resources/stable/2019-07-01/resources.json

```diff
@@ -4983,6 +5170,106 @@
           "description": "The template hash."
         }
       }
+    },
+    "WhatIfChange": {
+      "description": "Information about a single resource change predicted by What-If operation.",
+      "properties": {
+        "resourceId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource ID"
+        },
+        "changeType": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Type of change that will be made to the resource when the deployment is executed.",
+          "enum": [
+            "Create",
+            "Delete",
+            "Ignore",
+            "Deploy",
+            "NoChange",
+            "Modify"
+          ],
+          "x-ms-enum": {
+            "name": "ChangeType",
+            "modelAsString": false,
+            "values": [
+              {
+                "value": "Create",
+                "description": "The resource does not exist in the current state but is present in the desired state. The resource will be created when the deployment is executed."
+              },
+              {
+                "value": "Delete",
+                "description": "The resource exists in the current state and is missing from the desired state. The resource will be deleted when the deployment is executed."
+              },
+              {
+                "value": "Ignore",
+                "description": "The resource exists in the current state and is missing from the desired state. The resource will not be deployed or modified when the deployment is executed."
+              },
+              {
+                "value": "Deploy",
+                "description": "The resource exists in the current state and the desired state and will be redeployed when the deployment is executed. The properties of the resource may or may not change."
+              },
+              {
+                "value": "NoChange",
+                "description": "The resource exists in the current state and the desired state and will be redeployed when the deployment is executed. The properties of the resource will not change."
+              },
+              {
+                "value": "Modify",
+                "description": "The resource exists in the current state and the desired state and will be redeployed when the deployment is executed. The properties of the resource will change."
+              }
+            ]
+          }
+        },
+        "before": {
+          "readOnly": true,
+          "type": "object",
+          "description": "The snapshot of the resource before the deployment is executed."
+        },
+        "after": {
+          "readOnly": true,
+          "type": "object",
+          "description": "The predicted snapshot of the resource after the deployment is executed."
+        },
+        "delta": {
+          "readOnly": true,
+          "type": "object",
```

Comment:
>object [](start = 19, length = 6)

I'm intentionally leaving this as an opaque object for now to give us some leeway to alter the format if we discover any issues. This also matches the pattern we followed with the template itself, which is not modeled in the Swagger at all.

Reply:
It turns out we will need to model this after all. I pushed out an update.

---
In reply to: [325962761](https://github.com/Azure/azure-rest-api-specs/pull/7254#discussion_r325962761) [](ancestors = 325962761)